app/controllers/vendors_controller.rb

```diff
@@ -1,85 +1,44 @@
 class VendorsController < ApplicationController
   def index
-    uri = URI(rmapi_path)
-
     # Transform query params to what the EBSCO RM-API expects
-    uri.query = URI.encode_www_form(
+    query = URI.encode_www_form(
       search: params[:q],
       orderby: params[:orderby] || params[:q] ? 'relevance' : 'vendorname',
       count: params[:count] || 25,
       offset: params[:offset] || 1
     )
 
-    # Make the RM-API request and parse the response body
-    response = get_resource(uri)
-    data = Map JSON.parse(response.body)
-
-    # Return the list of vendors
-    if response.message === 'OK'
-      vendors = data.vendors.map do |vendor|
-        Vendor.new(vendor)
-      end
+    # Make the request for vendors from the RM API
+    data = rmapi.request(:get, "vendors?%{query}" % { query: query })
 
-      render jsonapi: vendors,
-             meta: { totalResults: data.totalResults },
-             status: :ok
+    if data
+      render jsonapi: data.vendors.map { |vendor| Vendor.new(vendor) },
+             meta: { totalResults: data.totalResults }
     else
-      render_rmapi_errors(data, response.code)
+      render jsonapi_errors: rmapi.errors,
+             status: rmapi.response.code
     end
   end
 
   def show
-    uri = URI(
-      "#{rmapi_path}/%{vendor_id}" % {
-        vendor_id: params[:id]
-      }
-    )
+    # Make the request for the vendor from the RM API
+    data = rmapi.request(:get, "vendors/%{id}" % { id: params[:id] })
 
-    # Make the RM-API request and parse the response body
-    response = get_resource(uri)
-    data = Map JSON.parse(response.body)
-
-    # Return the requested vendor
-    if response.message === 'OK'
-      vendor = Vendor.new(data)
-      render jsonapi: vendor, status: :ok
+    if data
+      render jsonapi: Vendor.new(data)
     else
-      render_rmapi_errors(data, response.code)
+      render jsonapi_errors: rmapi.errors,
+             status: rmapi.response.code
     end
   end
 
-  def get_resource(uri)
-    # Create the HTTP object
-    http = Net::HTTP.new(uri.host, uri.port)
-    http.use_ssl = true
+  private
 
-    # Create and send the request
-    http.get(
-      uri.request_uri,
-      {
-        "X-Api-Key" => configuration.api_key,
-        "Content-Type" => 'application/json',
-        "Accept" => 'application/json'
-      }
+  def rmapi
+    @rmapi ||= ::RmApiService.new(
```

Comment:
Is there a need to memoize this state?

Reply:
Yea, the service has two attributes we need access to when a request fails `rmapi.errors` and `rmapi.response.status`. We could have the request return the response, but then we'd have to repeat the parsing of JSON everywhere

Reply:
Rather than mutating the service, why not have the `success!` and `fail!` return `RmapiService::Response` objects. Then you get the best of both worlds.

Reply:
That sounds much better actually. Will start on those changes